account_financial_report/report/general_ledger.py

```diff
@@ -459,78 +459,86 @@ def _get_period_ml_data(
+                    if grouped_by:
+                        gen_ld_data[acc_id][grouped_by] = False
+                if acc_id in acc_prt_account_ids:
+                    item_ids = self._prepare_ml_items(move_line, grouped_by)
+                    for item in item_ids:
+                        item_id = item["id"]
+                        if item_id not in gen_ld_data[acc_id]:
+                            if grouped_by:
+                                gen_ld_data[acc_id][grouped_by] = True
+                            gen_ld_data[acc_id][item_id] = self._initialize_data(
+                                foreign_currency
+                            )
+                            gen_ld_data[acc_id][item_id]["id"] = item_id
+                            gen_ld_data[acc_id][item_id]["name"] = item["name"]
+                        gen_ld_data[acc_id][item_id][ml_id] = self._get_move_line_data(
+                            move_line
+                        )
+                        gen_ld_data[acc_id][item_id]["fin_bal"]["credit"] += move_line[
+                            "credit"
+                        ]
+                        gen_ld_data[acc_id][item_id]["fin_bal"]["debit"] += move_line[
+                            "debit"
+                        ]
+                        gen_ld_data[acc_id][item_id]["fin_bal"]["balance"] += move_line[
+                            "balance"
+                        ]
+                        if foreign_currency:
+                            gen_ld_data[acc_id][item_id]["fin_bal"][
+                                "bal_curr"
+                            ] += move_line["amount_currency"]
+                else:
+                    gen_ld_data[acc_id][ml_id] = self._get_move_line_data(move_line)
+                gen_ld_data[acc_id]["fin_bal"]["credit"] += move_line["credit"]
+                gen_ld_data[acc_id]["fin_bal"]["debit"] += move_line["debit"]
+                gen_ld_data[acc_id]["fin_bal"]["balance"] += move_line["balance"]
+                if foreign_currency:
+                    gen_ld_data[acc_id]["fin_bal"]["bal_curr"] += move_line[
+                        "amount_currency"
                     ]
-                    if foreign_currency:
-                        gen_ld_data[acc_id][item_id]["fin_bal"][
-                            "bal_curr"
-                        ] += move_line["amount_currency"]
-            else:
-                gen_ld_data[acc_id][ml_id] = self._get_move_line_data(move_line)
-            gen_ld_data[acc_id]["fin_bal"]["credit"] += move_line["credit"]
-            gen_ld_data[acc_id]["fin_bal"]["debit"] += move_line["debit"]
-            gen_ld_data[acc_id]["fin_bal"]["balance"] += move_line["balance"]
-            if foreign_currency:
-                gen_ld_data[acc_id]["fin_bal"]["bal_curr"] += move_line[
-                    "amount_currency"
-                ]
+                aml_ids.remove(move_line["id"])
```

Comment:
Do not modify the recordset you are iterating

Reply:
Corrected in new version